### PR TITLE
Refactor and implementation of captcha solver

### PR DESCRIPTION
**Pull request by boiled-water-tsar**
This PR implements the excellent captcha solver from https://github.com/boiled-water-tsar/eat-my-entire-trans-ass-andrew-bailey/issues/12

I'd like some more eyes on this as it's 1am and I don't know if this approach is a good idea or not anymore.

### CODE REVIEW

**Comment by lilyvxv** [comment]
looks good but you can’t really rely on status codes to determine if your request was successful or not as it still returns 200 even if you have invalid form data. it’s probably better to parse the response html with includes or proper html parsing

**Comment by graevy** [comment]
I'll look at it when I get back to my pc in ~6h but it looks very needed

File: requirements.txt [hunk 1]
```diff
@@ -6,3 +6,4 @@ python-dateutil==2.8.2
 requests==2.28.2
 six==1.16.0
 urllib3==1.26.15
+requests_toolbelt==0.10.1
```
Comment (sctsui) [inline requirements.txt:9]: minor "Cosmetic" suggestion: now that this pr adds a new package, the README instructions for dependencies installation is out of date. A catch-all 'pip install -r requirements.txt' can avoid this sort of docs mismatch in the future.
Also I found that the minimum python version is actually 3.10 due to python's 'match case' (I am running on a system with slightly older version)

**Comment by graevy** [comment]
i like it; do you want me to do a test suite for the audio captcha? do we want to pursue e.g. easyocr for visual solving? my main concern is a stronger captcha implementation but i think python ocr will require torch or something -- a 1gb dependency

also: the state dropdown menu is incorrectly coded without a default option rendering alabama an invalid option. i don't think we're up against much here lol

File: requirements.txt (hunk 1 above)
Comment (boiled-water-tsar) [reply requirements.txt:9]: made this an issue in itself so I can sort out my planning and thoughts https://github.com/boiled-water-tsar/eat-my-entire-trans-ass-andrew-bailey/issues/14

**Comment by boiled-water-tsar** [comment]
> i like it; do you want me to do a test suite for the audio captcha? do we want to pursue e.g. easyocr for visual solving? my main concern is a stronger captcha implementation but i think python ocr will require torch or something -- a 1gb dependency
> 
> also: the state dropdown menu is incorrectly coded without a default option rendering alabama an invalid option. i don't think we're up against much here lol

Either that or more easy of use / ease of contributing improvements e.g. making the ruff linter run before each pr and something similar for the test suite.

**Comment by boiled-water-tsar** [comment]
> looks good but you can’t really rely on status codes to determine if your request was successful or not as it still returns 200 even if you have invalid form data. it’s probably better to parse the response html with includes or proper html parsing.

absolutely and will be done in https://github.com/boiled-water-tsar/eat-my-entire-trans-ass-andrew-bailey/issues/15

**Pull request merged**